tests/test_other.py

```diff
@@ -9191,7 +9191,10 @@ def test_asan_no_stack_trace(self):
     ])
 
   def test_asan_pthread_stubs(self):
-    self.do_smart_test(test_file('other/test_asan_pthread_stubs.c'), emcc_args=['-fsanitize=address', '-sALLOW_MEMORY_GROWTH=1'])
+    self.do_smart_test(test_file('other/test_asan_pthread_stubs.c'), emcc_args=['-fsanitize=address', '-sALLOW_MEMORY_GROWTH'])
+
+  def test_asan_strncpy(self):
+    self.do_smart_test(test_file('other/test_asan_strncpy.c'), emcc_args=['-fsanitize=address', '-sALLOW_MEMORY_GROWTH'])
```

Comment:
I don't feel strongly, but perhaps all new libc methods could be tested in a single test (instead of adding a new one for each method)? Or will there be very few of these?

Reply:
This is not a new libc method.  Its a specific regression test for an asan false positive in strncpy.  Perhaps I should add a comment?

But also,  I'm bit surprised by this comment.  In general don't you think we should aim for the most precise tests we can?   "some libc function failed" is not a very actionable test failure.   As far I can test we mostly test libc function in their own tests today and I think that is good thing.

Reply:
Sorry, maybe I wasn't clear. This adds proper ASan integration for strncpy, but we may end up doing that for a lot of other libc methods. A single test for "ASan on libc string methods" might be good enough, to bundle them into groups at least.

I do agree individual tests are better in the way you said. However they do have minimal constant overhead. Precise tests when the tests are tiny sometimes feels excessive to me.

Reply:
How about I add a comment here and we can consider merging if we ever add more tests of this nature.

Reply:
Sgtm. No need for a comment even, just something to think about maybe if it recurs.